src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeCookieAction.cs

```diff
@@ -2,21 +2,80 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Rewrite.Internal.UrlActions
 {
     public class ChangeCookieAction : UrlAction
     {
-        public ChangeCookieAction(string cookie)
+        private readonly Func<DateTimeOffset> _timeSource;
+        private CookieOptions _cachedOptions;
+
+        public ChangeCookieAction(string name)
+            : this(name, () => DateTimeOffset.UtcNow)
         {
-            // TODO
-            throw new NotImplementedException("Changing the cookie is not implemented");
         }
 
+        // for testing
+        internal ChangeCookieAction(string name, Func<DateTimeOffset> timeSource)
+        {
+            _timeSource = timeSource;
+
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(nameof(name));
+            }
+
+            Name = name;
+        }
+
+        public string Name { get; }
+        public string Value { get; set; }
+
+        public string Domain { get; set; }
+
+        /// <summary>
+        /// Lifetime of cookie, expressed in minutes. Defaults to 0, a cookie that expires when the browser is closed.
+        /// </summary>
+        /// <returns></returns>
+        public int Lifetime { get; set; }
```

Comment:
Well in that case I guess its fine. That value is logically slightly different. Its represents the date when it expires where as this value represents how long it can exist. Starting to like the original name Lifetime 😄